### PR TITLE
Colors in graph

### PR DESCRIPTION
**Pull request by natanrolnik**
### Short description 📝

For large graphs with different types of dependencies and targets, visualization was hard. I had the idea to add different types of shapes and colors that help visualizing them.

I've updated the docs accordingly (including the change related to #1762), adding a sample image so users can understand what the graph will do, and also adding a legend showing what each shape/color means.

I needed to fork [GraphViz under Tuist's org](https://github.com/tuist/GraphViz/), to [allow adding support to more shapes](https://github.com/SwiftDocOrg/GraphViz/pull/7), and also to include a fix to filled shapes ([which hasn't been merged yet](https://github.com/SwiftDocOrg/GraphViz/pull/6) to the main GraphViz repo).

### Usage

Tuist now will generate a graph like this:

GraphExample

Users who want to keep the old plain style, can pass the `--simple` flag when generating a graph.

### CODE REVIEW

**Review by pepibumur — APPROVED** [review]
Don't forget to update the CHANGELOG.md :).

File: Sources/TuistKit/Commands/GraphCommand.swift [hunk 1]
```diff
@@ -37,6 +37,12 @@ struct GraphCommand: ParsableCommand {
     )
     var layoutAlgorithm: GraphViz.LayoutAlgorithm = .dot
 
+    @Flag(
```
[the remark this answers is not in the document]
Comment (fortmarek) [reply Sources/TuistKit/Commands/GraphCommand.swift:40]: +1 - I don't see why some people would not want this awesome new output format and it's gonna make the implementation much more streamlined ✨

**Review by fortmarek — APPROVED** [review]
This looks really great, well done 👏

**Comment by jeroenleenarts** [comment]
Grouping is mentioned. This would work great with a subGraph in dot syntax.

Grouped by team

**Review by jeroenleenarts — APPROVED** [review]

**Review by laxmorek — APPROVED** [review]

**Pull request merged**